addons/udes_stock/models/stock_move_line.py

```diff
@@ -654,5 +657,17 @@ def _validate_location_dest(self):
         for picking in self.mapped('picking_id'):
             if not picking.is_valid_location_dest_id(location=location):
                 raise ValidationError(
-                    _("The location '%s' is not a child of the picking destination "
-                      "location '%s'" % (location.name, picking.location_dest_id.name)))
+                    _("The location '%s' is not a child of the picking "
+                      "destination location '%s'" %
+                      (location.name, picking.location_dest_id.name)))
+
+            if picking.picking_type_id.u_drop_location_constraint == 'enforce' \
+                    and picking.state == 'assigned':
+                mls = self.filtered(lambda ml: ml.picking_id == picking)
```

Comment:
Also as this effectively doing a self.groupby('picking_id') 
We are going to be adding edi as a dependency of udes_stock till that functionality is factored out.
Might be worth looking at.

Reply:
Not sure I understand what you want here.
Even if you rely on `groupby`, you still to select the group with the picking id you're interested in, no?
What's the benefit?
The `filtered` call is pretty clear to me.

Reply:
you can do 
`for picking, mls in self.groupby('picking_id'):`
I personally think its a bit neater but I don't have strong feeling about this just thought I'd mention it.

Reply:
Ah, makes sense, I'll add a TODO.